processing/src/main/java/org/apache/druid/segment/writeout/FileWriteOutBytes.java

```diff
@@ -59,7 +60,12 @@ private void flushIfNeeded(int bytesNeeded) throws IOException
   public void flush() throws IOException
   {
     buffer.flip();
-    Channels.writeFully(ch, buffer);
+    try {
+      Channels.writeFully(ch, buffer);
+    }
+    catch (IOException e) {
+      throw new IOE(e, "Failed to write to file: %s. Space needed: %d", file.getAbsolutePath(), writeOutBytes);
```

Comment:
From the exception, it seems like that the `writeOutBytes` is the space deficit due to which the ingestion is failing, whereas the actual deficit for the job to complete can be much more than that right?
Should the error message omit the information `Space needed: %d` or be reworded such that the user knows that `writeOutBytes` more are required to complete the current operation (and not the ingestion)?

Reply:
I think we can't guess how much many more bytes are needed as that is dependent on when the user will call the close method.